### PR TITLE
Us008 view training by band

### PR DESCRIPTION
**Pull request by ieuanbyers**
Reopening PR after updating job-roles to link to training courses

### CODE REVIEW

File: src/main/java/com/kainos/ea/controller/TrainingController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+@Api("Training")
```
Comment (martin-doherty1) [inline src/main/java/com/kainos/ea/controller/TrainingController.java:20]: could change the name to be consistent like: @Api("Backend API Bands")

File: src/test/java/com/kainos/ea/integration/TrainingIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    );
+
+    @Test
+    void getTrainingByBand_shouldReturnListOfTrainingCourses() {
+        int id = 1;
+        List<TrainingRequest> response = APP.client().target("http://localhost:8080/api/training/" + id)
+                .request()
+                .get(List.class);
+
+        Assertions.assertTrue(response.size() > 0);
```
Comment (martin-doherty1) [inline src/test/java/com/kainos/ea/integration/TrainingIntegrationTest.java:30]: do an assertion to check a value within the list returned

File: src/main/java/com/kainos/ea/controller/TrainingController.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public TrainingController() {
+        trainingService = new TrainingService(new TrainingDao(), new DatabaseConnector());
+    }
+
+    public TrainingController(TrainingService trainingService) {
+        this.trainingService = trainingService;
+    }
+
+    @GET
+    @Path("/training/{id}")
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/controller/TrainingController.java:35]: could we rename this to be bandId?
```suggestion
    @Path("/training/{bandId}")
```

File: src/main/java/com/kainos/ea/dao/TrainingDao.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+
+public class TrainingDao {
+
+    public List<TrainingRequest> getTrainingByBand(int bandId, Connection c) throws SQLException {
+        Statement st = c.createStatement();
+
+        ArrayList<TrainingRequest> trainingList = new ArrayList<>();
+
+        String selectStatement = "SELECT Training.name, Training.link, Training_Categories.name FROM Training JOIN Training_Bands USING(training_id) JOIN Training_Categories USING(training_category_id) JOIN Bands USING(band_id) WHERE band_id = " + bandId;
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/dao/TrainingDao.java:19]: could we update this to select the band name as well? That way we don't need to make 2 HTTP requests to populate this page

File: src/main/java/com/kainos/ea/controller/BandController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.sql.SQLException;
+
+@Api("Backend API Bands")
+@Path("/api")
+public class BandController {
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/controller/BandController.java:22]: If we add the band name to the training response I think we could remove these

**Review by ccurran01 — COMMENTED** [review]
Nice work. Just a few suggestions

**Review by ccurran01 — APPROVED** [review]

**Review by malachi-mccarney — APPROVED** [review]
Looks good

**Pull request merged**